### PR TITLE
Update staging3Test for informatici

### PR DESCRIPTION
**Pull request by mwithi**
…ratoryForPrintUsingGET in order to return a list of LabWithRowsDTO

### CODE REVIEW

File: src/main/java/org/isf/lab/service/LabIoOperations.java [hunk 1]
```diff
@@ -104,6 +104,33 @@ public List<Laboratory> getLaboratory(String exam, LocalDateTime dateFrom, Local
...
+	 * @param dateFrom - the lower date for the range
+	 * @param dateTo - the highest date for the range
+	 * @param patient - the object of patient 
+	 * @return the list of {@link Laboratory}s 
+	 * @throws OHServiceException
+	 */
+	public List<Laboratory> getLaboratory(String exam, LocalDateTime dateFrom, LocalDateTime dateTo, Patient patient) throws OHServiceException {
+		List<Laboratory> laboritories = new ArrayList<>();
+		
+		if(exam.equals("") && patient != null) {
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/lab/service/LabIoOperations.java:119]:
```suggestion
		if (exam.equals("") && patient != null) {
```

Comment (mwithi) [reply src/main/java/org/isf/lab/service/LabIoOperations.java:119]: sorry, I will fix that

File: src/main/java/org/isf/lab/service/LabIoOperations.java [hunk 2]
```diff
@@ -104,6 +104,33 @@ public List<Laboratory> getLaboratory(String exam, LocalDateTime dateFrom, Local
...
+			laboritories = repository.findByLabDateBetweenAndExamDescriptionAndPatientCode(dateFrom, dateTo, exam, patient.getCode());
+		}
+		if( exam.equals("") && patient == null ) {
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/lab/service/LabIoOperations.java:122]:
```suggestion
		if ( exam.equals("") && patient == null ) {
```

File: src/main/java/org/isf/lab/service/LabIoOperations.java [hunk 3]
```diff
@@ -104,6 +104,33 @@ public List<Laboratory> getLaboratory(String exam, LocalDateTime dateFrom, Local
...
+			laboritories = repository.findByLabDateBetweenAndExam_DescriptionOrderByLabDateDesc(dateFrom, dateTo, exam);
+		}
+		if(patient != null && exam.equals("")) {
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/lab/service/LabIoOperations.java:125]:
```suggestion
		if (patient != null && exam.equals("")) {
```

Comment (dbmalkovsky) [reply src/main/java/org/isf/lab/service/LabIoOperations.java:125]: How is line 125 different than line 119 (other than the ordering of the comparisons?

File: src/main/java/org/isf/lab/service/LabIoOperations.java [hunk 4]
```diff
@@ -104,6 +104,33 @@ public List<Laboratory> getLaboratory(String exam, LocalDateTime dateFrom, Local
...
+			laboritories = repository.findByLabDateBetweenAndPatientCode(dateFrom, dateTo, patient.getCode());
+		}
+		if(patient == null && exam.equals("")) {
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/lab/service/LabIoOperations.java:128]:
```suggestion
		if (patient == null && exam.equals("")) {
```

Comment (dbmalkovsky) [reply src/main/java/org/isf/lab/service/LabIoOperations.java:128]: How is line 128 different than line 122?

File: src/main/java/org/isf/lab/service/LabIoOperations.java [hunk 5]
```diff
@@ -104,6 +104,33 @@ public List<Laboratory> getLaboratory(String exam, LocalDateTime dateFrom, Local
...
+			laboritories= repository.findByCreatedDateBetweenOrderByLabDateDesc(dateFrom, dateTo);
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/lab/service/LabIoOperations.java:129]:
```suggestion
			laboritories = repository.findByCreatedDateBetweenOrderByLabDateDesc(dateFrom, dateTo);
```

File: src/main/java/org/isf/lab/service/LabIoOperations.java (hunk 3 above)
Comment (mwithi) [reply src/main/java/org/isf/lab/service/LabIoOperations.java:125]: I don't know... let me ask (it was also on informatici repo the same way)

**Pull request merged**